kafkaclient/client_test.go

```diff
@@ -697,6 +697,139 @@ func TestAzureEventHubsCloud(t *testing.T) {
+		dockerKafka.WithBrokers(1))
+	require.NoError(t, err)
+
+	addresses := make([]string, 0, len(kafkaContainer.Ports))
+	for i := 0; i < len(kafkaContainer.Ports); i++ {
+		addresses = append(addresses, fmt.Sprintf("localhost:%s", kafkaContainer.Ports[i]))
+	}
+	c, err := New("tcp", addresses, Config{ClientID: "some-client", DialTimeout: 5 * time.Second})
+	require.NoError(t, err)
+
+	var (
+		noOfMessages = 10
+		ctx, cancel  = context.WithCancel(context.Background())
+		tc           = testutil.NewWithDialer(c.dialer, c.network, c.addresses...)
+	)
+
+	t.Cleanup(cancel)
+
+	// Check connectivity and try to create the desired topic until the brokers are up and running (max 30s)
+	require.NoError(t, c.Ping(ctx))
+	require.Eventually(t, func() bool {
+		err := tc.CreateTopic(ctx, t.Name(), 1, 1) // partitions = 1, replication factor = 1
+		if err != nil {
+			t.Logf("Could not create topic: %v", err)
+		}
+		return err == nil
+	}, defaultTestTimeout, time.Second)
+
+	// Check that the topic has been created with the right number of partitions
+	var topics []testutil.TopicPartition
+	require.Eventually(t, func() bool {
+		topics, err = tc.ListTopics(ctx)
+		success := err == nil && len(topics) == 1
+		if !success {
+			t.Logf("List topics failure %+v: %v", topics, err)
+		}
+		return success
+	}, defaultTestTimeout, time.Second)
+	require.Equal(t, []testutil.TopicPartition{
+		{Topic: t.Name(), Partition: 0},
+	}, topics)
+
+	// Produce X messages in a single batch
+	producerConf := ProducerConfig{
+		ClientID:     "producer-01",
+		WriteTimeout: 5 * time.Second,
+		ReadTimeout:  5 * time.Second,
+		Logger:       newKafkaLogger(t, false),
+		ErrorLogger:  newKafkaLogger(t, true),
+	}
+	p, err := c.NewProducer(producerConf)
+	require.NoError(t, err)
+	publishMessages(ctx, t, p, noOfMessages)
+
+	// Starting consumers with group-01 and FirstOffset
+	var (
+		// The ticker is used so that the test won't end as long as we keep getting messages since the consumers
+		// will reset the ticker each time they receive a message
+		tickerMu    sync.Mutex
+		tickerReset = 10 * time.Second
+		ticker      = time.NewTicker(30 * time.Second)
+	)
+	consumerConf := ConsumerConfig{
+		GroupID:             "group-01",
+		StartOffset:         FirstOffset,
+		CommitInterval:      time.Second, // to make the test faster instead of committing each single message
+		FetchBatchesMaxWait: 10 * time.Second,
+		Logger:              newKafkaLogger(t, false),
+		ErrorLogger:         newKafkaLogger(t, true),
+	}
+	consume := func(c *Consumer, id string, count int32) []Message {
+		messages := make([]Message, 0, count)
+		for count > 0 {
+			msg, err := c.Receive(ctx)
+			if errors.Is(err, io.EOF) || errors.Is(err, context.Canceled) {
+				t.Logf("Closing %s: %v", id, err)
+				return messages
+			}
+			require.NoError(t, err)
+			t.Logf("Got a message on %s", id)
+			tickerMu.Lock()
+			ticker.Reset(tickerReset)
+			tickerMu.Unlock()
```

Comment:
Didn't get how this is being used. We are resetting the ticker after every consumption, But not using the channel provided by the ticker anywhere?

Reply:
Removed this logic